### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by deerawj**
### Description

Fixed some simple typos in comments and documentation.
Corrected an image filename and updated the reference to image.

While the typos were programmatically found, all of them were manually checked.
I also tried to read as much context as possible; so for example the instances where words are intentionally misspelled (`labeledby`, `traget`, `mistypeVarible`) are skipped. 

### Motivation

I originally used this repo to correct false positives in a spell checker I'm working on; but I noticed that there are some actual typos, so...

### Additional details

None

### Related issues and pull requests

None

### CODE REVIEW

File: files/en-us/learn/tools_and_testing/understanding_client-side_tools/introducing_complete_toolchain/index.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ For our sample project, we'll be using a toolchain specifically designed to aid
 
 ## Checking prerequisites
 
-You should have most of the softwares already if you've been following along with the previous chapters. Here's what you should have before proceeding to the real setup steps. They only need to be done once and you don't need to repeat these again for future projects.
+You should have most of the software already if you've been following along with the previous chapters. Here's what you should have before proceeding to the real setup steps. They only need to be done once and you don't need to repeat these again for future projects.
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/understanding_client-side_tools/introducing_complete_toolchain/index.md:67]: We often say "softwares" and I don't consider it a typo personally. Technically it's "pieces of software" but that's too verbose for our purpose.

**Review by Josh-Cena — APPROVED** [review]
Awesome work. FYI, we just recently adopted cSpell and it's giving us weekly spelling reports, e.g. https://github.com/mdn/content/issues/36323

We have a lot of existing typos, and they will be addressed in https://github.com/mdn/content/pull/36242 and its many siblings that remain open.

**Review by bsmth — APPROVED** [review]
Well done. There's one comment from Josh outstanding, but I think we can merge after you've had a look.

File: files/en-us/web/api/customstateset/index.md [hunk 2]
```diff
@@ -485,7 +485,7 @@ Click the element to see a different border being applied as the state changes.
 
 {{EmbedLiveSample("Non-boolean internal states", "100%", 50)}}
 
-## Comptability with `<dashed-ident>` syntax
+## Comptibility with `<dashed-ident>` syntax
```
Comment (estelle) [inline files/en-us/web/api/customstateset/index.md:488]:
```suggestion
## Compatibility with `<dashed-ident>` syntax
```

**Review by estelle — APPROVED** [review]
one comment. Otherwise looks good. 
Thank you. Also, congratulations on your first MDN content PR. 
Welcome to the team!

File: files/en-us/learn/tools_and_testing/understanding_client-side_tools/introducing_complete_toolchain/index.md (hunk 1 above)
Comment (Josh-Cena) [reply files/en-us/learn/tools_and_testing/understanding_client-side_tools/introducing_complete_toolchain/index.md:67]:
```suggestion
You should have most of the softwares already if you've been following along with the previous chapters. Here's what you should have before proceeding to the real setup steps. They only need to be done once and you don't need to repeat these again for future projects.
```

**Pull request merged**